lib/neo4j.rb

```diff
@@ -1,8 +1,6 @@
 require 'neo4j/version'
 
 require 'neo4j-core'
-require 'neo4j/core/cypher_session'
-require 'neo4j/core/query' # From the `neo4j-core` gem
```

Comment:
I got problems with those requires. They shouldn't be necessary as they are already required in the line above.

Reply:
Ideally I'd like to bring the code from this PR in as a patch for the current release of the `neo4j` gem.  Does this change work correctly against that (as opposed to `master` of `neo4j-core`)?

Reply:
@cheerfulstoic are you specifically interested in the dynamic loading or the converter fix? If it is the former I still have to do some work in railtie which still loads everything. Now I see I should've done 2 PRs, but bear with me. I'll let you know as well if it can be applied to the current release.

Reply:
The dynamic loading would probably technically be a minor version bump (backward-compatible change), though I could also see it as a fix because only what is needed is being loaded.  Basically I just wanted to make sure that the `BigDecimnalConverter` fix could be made into a 9.2.4 release so that people could get the fix without needing to update to the major version